### PR TITLE
Add script to pre-process raw release notes

### PR DESCRIPTION
**Pull request by andrewor14**
Summary: This commit adds a helper script to pre-process the raw release notes produced by github. This script produces a template with the standard sections and pre-sorts all commits into categories based on github labels and keywords in the commit titles.

```
python scripts/clean_release_notes.py raw_release_notes.txt
```

Test Plan: Manual testing on v0.7.0 release notes

### CODE REVIEW

File: scripts/clean_release_notes.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        label = pr_number_to_label[pr_number]
+        if label == "topic: not user facing":
+            return None
+        if label in GITHUB_LABEL_TO_CATEGORY:
+            return GITHUB_LABEL_TO_CATEGORY[label]
+    elif any(x in commit_line.lower() for x in ["revert", "version.txt"]):
+        return None
+    elif any(x in commit_line.lower() for x in ["doc", "readme", "tutorial", "typo", "example", "spelling"]):
+        return "Documentation"
+    elif any(x in commit_line.lower() for x in ["test", "lint", " ci", "nightl"]):
```
Comment (jerryzh168) [inline scripts/clean_release_notes.py:167]:
```suggestion
    elif any(x in commit_line.lower() for x in ["test", "lint", " ci", "nightly"]):
```

Comment (andrewor14) [reply scripts/clean_release_notes.py:167]: This is on purpose actually :), so we match both nightly and nightlies (this appeared in some commit titles)

File: scripts/clean_release_notes.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+#  This script automatically cleans up the raw release notes
+#  generated by github by doing an initial pass to sort the
+#  commits. The output still requires manual reviewing.
+#
+#  This script uses PyGithub. If you don't have it yet, please
+#  install it using
+#
+#    pip install PyGithub
+#
+#  We expect the following format for the input release notes:
```
Comment (jerryzh168) [inline scripts/clean_release_notes.py:17]: add an example output as well?

**Review by jerryzh168 — APPROVED** [review]

**Comment by andrewor14** [comment]
Thanks, merging this!

**Pull request merged**